python_modules/dagster/dagster/_core/remote_representation/external_data.py

```diff
@@ -445,11 +447,19 @@ def __new__(
         execution_timezone: Optional[str],
         description: Optional[str] = None,
         default_status: Optional[DefaultScheduleStatus] = None,
+        asset_selection: Optional[AssetSelection] = None,
     ):
         cron_schedule = check.inst_param(cron_schedule, "cron_schedule", (str, Sequence))
         if not isinstance(cron_schedule, str):
             cron_schedule = check.sequence_param(cron_schedule, "cron_schedule", of_type=str)
 
+        if asset_selection is not None:
+            check.opt_inst_param(asset_selection, "asset_selection", AssetSelection)
+            check.invariant(
+                is_whitelisted_for_serdes_object(asset_selection),
```

Comment:
when is this not true?

Reply:
I'm actually not sure. I'm cargo-culting here from the analogous `asset_selection` arg for `SensorSnap`, which already existed and used this check.

Reply:
On further investigation it looks like we've left the door open to have unserializable `AssetSelection` subclasses, but currently all `AssetSelection` subclasses are whitelisted. So in the current codebase it will never not be true, but in theory an `AssetSelection` subclass could be introduced in the future that makes it `False`.

But if you go through `{Sensor,Schedule}Snap.from_def` it will be converted to serializable every time with `to_serializable_asset_selection`:

```
    def to_serializable_asset_selection(self, asset_graph: BaseAssetGraph) -> "AssetSelection":
        return KeysAssetSelection(selected_keys=list(self.resolve(asset_graph)))
```

Reply:
Great thanks for the investigation and explanation.